### PR TITLE
change import syntax

### LINKED ISSUE #215: import all functions from file
How to import all functions from file? (I an writing math lib test for jou)
Or, jou doesn't support it yet?

### PR DESCRIPTION
**Pull request by Akuli**
Fixes #215

### CODE REVIEW

File: self_hosted/typecheck.jou [hunk 1]
```diff
@@ -78,13 +70,13 @@ class TypeContext:
 #def typecheck_stage1_create_types(ctx: TypeContext*, file: AstFile*) -> ExportSymbol*:
 #    assert(False)
 
-def type_from_ast(ctx: TypeContext*, ast_type: AstType*) -> Type*:
-    result = type_or_void_from_ast(ctx, ast_type)
+def type_fromast(ctx: TypeContext*, ast_type: AstType*) -> Type*:
```
Comment (Akuli) [inline self_hosted/typecheck.jou:73]: this was not supposed to happen

File: src/main.c [hunk 2]
```diff
@@ -407,25 +399,6 @@ static void add_imported_symbols(struct CompileState *compst)
...
-Check whether each import statement in AST actually imported something.
-
-This is trickier than you would expect, because multiple passes over
-the AST look at the imports, and any of them could provide the symbol to import.
-*/
-static void check_for_404_imports(const struct CompileState *compst)
-{
-    for (struct FileState *fs = compst->files.ptr; fs < End(compst->files); fs++) {
-        for (const AstToplevelNode *imp = fs->ast; imp->kind == AST_TOPLEVEL_IMPORT; imp++) {
-            if (!imp->data.import.found) {
```
Comment (Akuli) [inline src/main.c:420]: I wonder if we still need `found` boolean

File: src/parse.c [hunk 3]
```diff
@@ -1007,11 +964,13 @@ AstToplevelNode *parse(const Token *tokens, const char *stdlib_path)
...
-        if (is_keyword(tokens, "from"))
-            parse_import(&tokens, stdlib_path, &result);
-        else
-            Append(&result, parse_toplevel_node(&tokens));
+        Append(&result, parse_toplevel_node(&tokens, stdlib_path));
     } while (result.ptr[result.len - 1].kind != AST_TOPLEVEL_END_OF_FILE);
+
+    // This simplifies the compiler: it's easy to loop through all imports of the file.
+    for (int i = 1; i < result.len; i++)
+        if (result.ptr[i-1].kind != AST_TOPLEVEL_IMPORT && result.ptr[i].kind == AST_TOPLEVEL_IMPORT)
```
Comment (Akuli) [inline src/parse.c:972]: for fun this could be written with pointer magic instead of indexing :)

**Comment by littlewhitecloud** [comment]
Will it take more memory?

**Comment by Akuli** [comment]
Not noticably. LLVM stuff (creating LLVM IR, optimizing IR, writing IR to object files) takes up much more memory than the rest of the compilation, and this doesn't affect the LLVM IR much.

**Pull request merged**